luxai2021/game/game_map.py

```diff
@@ -82,14 +82,16 @@ def _rng():
             rng = js_rng(math.floor(random.random() * 1e9))
 
         size = mapSizes[math.floor(rng.random() * len(mapSizes))]
-        if "width" not in self.configs:
```

Comment:
While this is a solution, I think the root cause is that self.configs probably points to the same underlying dictionary as the input argument config. Meaning that when it's edited here, it's editing the base default config as well.

How about a solution at config assignment to deep copy it:
self.config = dict(config)

I think this solution might be a bit better, that way it has a real copy of the config that isn't going to change.

Reply:
Yes, it points to the same dictionary, but I see it as a good thing. By having several copies, we open the door for several _different_ copies, which I don't see how it is any good. Having different configs in different classes sounds like an opening for a trouble. Instead, treating the config as read-only in such internal places is the preferred way I think. WDUT?

Reply:
Making the config read-only is a good idea :+1: With the proposed change the config dict is not edited anymore and the `Game` `reset()` will now generate a map with random size each time.